src/words/fast.jl

```diff
@@ -135,13 +151,32 @@ end
 
 Matches a set of atomic tokens, such as `...`, which should always be treated
 as a single token, regardless of word boundaries.
+Tokens to be matched can also be given in Regex format or an array of regex and str, where str 
+will be replaced with found regex and then tokenized.
 """
 function atoms(ts, as)
   for a in as
-    lookahead(ts, a) || continue
-    flush!(ts, String(a))
-    ts.idx += length(a)
-    return true
+    if typeof(a) == Regex || typeof(a) == Tuple{Regex,String}
```

Comment:
Break this into methods, and take advatage of dispatch .

Reply:
@oxinabox I have done it this way thinking if the user is giving a heterogeneous array of regex rule and a simple string token. Breaking it will increase user overhead to call same function twice after separating the two.  
What are your thoughts, should we still break it considering this case?